freqtrade/rpc/external_message_consumer.py

```diff
@@ -336,16 +366,58 @@ def _consume_analyzed_df_message(self, producer_name: str, message: WSMessageSch
 
         pair, timeframe, candle_type = key
 
+        if df.empty:
+            logger.debug(f"Received Empty Dataframe for {key}")
+            return
+
         # If set, remove the Entry and Exit signals from the Producer
         if self._emc_config.get('remove_entry_exit_signals', False):
             df = remove_entry_exit_signals(df)
 
-        # Add the dataframe to the dataprovider
-        self._dp._add_external_df(pair, df,
-                                  last_analyzed=la,
-                                  timeframe=timeframe,
-                                  candle_type=candle_type,
-                                  producer_name=producer_name)
+        logger.debug(f"Received {len(df)} candle(s) for {key}")
+
+        if len(df) >= 999:
+            # This is a full dataframe
+            # Add the dataframe to the dataprovider
+            self._dp._add_external_df(
+                pair,
+                df,
+                last_analyzed=la,
+                timeframe=timeframe,
+                candle_type=candle_type,
+                producer_name=producer_name
+            )
+
+        elif len(df) < 999:
+            # This is n single candles
+            # Have dataprovider append it to
+            # the full datafame. If it can't,
+            # request the missing candles
+            did_append, n_missing = self._dp._add_external_candle(
+                pair,
+                df,
+                last_analyzed=la,
+                timeframe=timeframe,
+                candle_type=candle_type,
+                producer_name=producer_name
+            )
+
+            if not did_append:
+                logger.debug("Holes in data or no existing df, "
```

Comment:
I think we should log this in info or warning mode.

i understand you made this for debugging purposes - but it's a pretty important info - as it can hint at "your producer is too slow, and we're gonna make that worse by requesting it all again" ...